contracts/liquidation/priceOracle/chainlink/ChainLinkPriceOracle.sol

```diff
@@ -0,0 +1,150 @@
+  */
+  function updateAggregatorProxyData(
+    address[] calldata tokens,
+    address[] calldata quoteEthAddresses,
+    address[] calldata quoteUsdAddresses
+  ) external onlyAdmin {
+
+    require(
+      tokens.length == quoteEthAddresses.length &&
+      tokens.length == quoteUsdAddresses.length,
+      'invalid length data'
+    );
+
+    for(uint256 i = 0; i < tokens.length; i++) {
+      _tokenData[tokens[i]] = AggregatorProxyData({
+        quoteEth: quoteEthAddresses[i],
+        quoteUsd: quoteUsdAddresses[i]
+      });
+    }
+  }
+  /**
+  *  @dev Get conversion rate from src to dest token given amount
+  *   For chainlink, amount is not needed
+  *   Fetch rates using both eth and usd as quote, then take the average
+  */
+  function conversionRate(
+    address src,
+    address dest,
+    uint256 /* amount */
+  )
+    external override view returns(uint256 rate)
+  {
+    if (dest == address(ETH_TOKEN_ADDRESS)) {
+      return getRateOverEth(src);
+    }
+
+    if (src == address(ETH_TOKEN_ADDRESS)) {
+      rate = getRateOverEth(dest);
+      if (rate > 0) rate = PRECISION.mul(PRECISION).div(rate);
+      return rate;
+    }
+
+    uint256 srcRate;
+    uint256 destRate;
+
+    uint256 rateQuoteEth;
+    uint256 rateQuoteUsd;
+
+    // get rate from eth quote
+    srcRate = getRateOverEth(src);
+    if (srcRate > 0) {
+      destRate = getRateOverEth(dest);
+      if (destRate > 0) {
+        rateQuoteEth = PRECISION.mul(srcRate).div(destRate);
+      }
+    }
+
+    // get rate from usd quote
+    srcRate = getRateOverUsd(src);
+    if (srcRate > 0) {
+      destRate = getRateOverUsd(dest);
+      if (destRate > 0) {
+        // update new rate if it is higher
+        rateQuoteUsd = PRECISION.mul(srcRate).div(destRate);
+      }
+    }
+
+    if (rateQuoteEth == 0) {
+      rate = rateQuoteUsd;
+    } else if (rateQuoteUsd == 0) {
+      rate = rateQuoteEth;
+    } else {
+      // average rate over eth and usd
+      rate = rateQuoteEth.add(rateQuoteUsd).div(2);
+    }
+  }
+
+  function getTokenAggregatorProxyData(address token)
+    external view returns (
+      address quoteEth,
+      address quoteUsd
+    )
+  {
+    (quoteEth, quoteUsd) = (_tokenData[token].quoteEth, _tokenData[token].quoteUsd);
+  }
+
+  function getRateOverEth(address token) public view returns (uint256 rate) {
+    int answer;
+    IChainLinkAggregatorProxy proxy = IChainLinkAggregatorProxy(_tokenData[token].quoteEth);
+    if (proxy != IChainLinkAggregatorProxy(0)) {
+      (, answer, , ,) = proxy.latestRoundData();
+    }
+    if (answer < 0) return 0;
+    rate = uint256(answer);
+  }
+
+  function getRateOverUsd(address token) public view returns (uint256 rate) {
+    int answer;
+    IChainLinkAggregatorProxy proxy = IChainLinkAggregatorProxy(_tokenData[token].quoteUsd);
+    if (proxy != IChainLinkAggregatorProxy(0)) {
+      (, answer, , ,) = proxy.latestRoundData();
```

Comment:
I check from chainlink aave/usd
https://etherscan.io/address/0x547a514d5e3769680Ce22B2361c10Ea13619e8a9#code

So the rate is 43648600684 (436$) with decimal = **8**
This rate is not fit with our conversion rate (base decimals = **18**)

Reply:
If src or dest is ETH, rate is with quote eth, so it is fine
If src and dest are not eth, rate will be (src/X) / (dest/X), so it doesn't matter the decimal of X.
`PRECISION.mul(srcRate).div(destRate)` this will change to our current decimals 18.

But I have updated both get rate func to return rate in decimals 18, so less confusing